### ISSUE TITLE
텐서플로우 강의 관한내용

### ISSUE THREAD

**Issue by MintyStory** [codingeverybody/codingyahac#984]
질문 1 - 이 프로그래밍 언어가 파이썬 프로그래밍 언어인가요?

독립 = 데이터[['온도']]
종속 = 데이터[['판매량']]
print(독립.shape, 종속.shape)

모델을 만든다

X = tf.keras.layers.Input(shape=[1])
Y = tf.keras,layers,Dense(1)(X)
model = tf.keras.models.model(X, Y)
model.compile(loss='mse')

모델을 학습시킨다

model.fit(독립, 종속, epochs=10000)

모델을 이용한다.
model.predict(독립)
model.predict([[15]])

질문2 - 파이썬 프로그래밍 언어
머신러닝 프로그래머 개발자가 되려면 파이썬 언어만 좀 알고 프로젝트좀 몇개 만들어서 준비해 있으면 될까요?

질문 3 - 질문 1에서 보이는 코드중에 tf.keras / Dense / oss='mse'이 세부분을 한글말로 어떤걸 의미하는건지랑 어떻게 이해를 해야할지요

**Comment by podong28** [comment]
질문1 - 파이썬 맞습니다.
질문2 - 언어 문법을 몰라서 머신러닝이 안된다기 보다 왜 이런 함수,파라미터들을 사용하는지 이해를 못하는경우가 많더군요.
질문3 - 다른고수님이 답변을... keras는 케라스 기능을 불러오는듯...

**Comment by RayleighKim** [comment]
>질문 3 - 질문 1에서 보이는 코드중에 tf.keras / Dense / oss='mse'이 세부분을 한글말로 어떤걸 의미하는건지랑 어떻게 이해를 해야할지요

1. 조금 더 이어지는 강의를 더 살펴보시기를 추천합니다.
2. tf.keras 이전에 import tensorflow as tf가 있었을거에요.  tf.keras는 텐서플로우 라이브러리 안에 준비 된 케라스를 사용하겠다는 의미를 지니게 됩니다.
3. 선형회귀-뉴럴넷 공부를 하게되면 훨씬 부드럽게 와닿으실텐데요.   앞전레이어의 노드들로 부터 전부 연결지어 만드는 레이어를 fully connected layer 또는 densely connected layer라고 합니다. dense는 이러한 레이어를 규현할 때 사용하는 코드입니다.
4. 연결주의의 머신러닝에서 지도학습알고리즘은 다음과 같은 절차로 학습을 시킵니다.
    1. Y와 X의 관계 구조를 잡음
    2. X를 이용해 Y예측값 생성
    3. Loss계산 : Y예측값과 Y실제값을 비교하기
    4. Loss를 줄이도록 관계 구조내부의 가중치 업데이트
5. 즉, 모델을 학습시키려면 loss를 정의해야 하고.. 회귀문제를 풀기 위하여 mean squared error를 사용하겠다는 의미가 됩니다.

**Comment by MintyStory** [comment]
그렇군요 이제 이해가 잘 되네요 감사합니다 선생님
